### DISCUSSION TITLE
Github integration of mermaidjs

### DISCUSSION THREAD

**Discussion by emilien-puget** [community/community#60159] [New to GitHub] [answered]
Is the version used of mermaidjs updated regularly ? the current version seems to be a little bit oudated.

**Comment by F4NT0** [comment]
Yes, the Mermaid.js from Github Currently is V10.2.4 like the [official website](https://mermaid.js.org/) but not all features work completely on Github.

One Working example is using mindmap, but the characters don't work on Github

```mermaid
mindmap
    Root
        A
        ::icon(fa fa-book)
        B(B)
        ::icon(mdi mdi-skull-outline)
```

The code from the example:

```
mindmap
    Root
        A
        ::icon(fa fa-book)
        B(B)
        ::icon(mdi mdi-skull-outline)
```

Other example who doesn't work on Github is the use of Zenuml like shown in this [page](https://mermaid.js.org/syntax/zenuml.html)

It give's the folliwing error:

```mermaid
zenuml
    title Demo
    Alice->John: Hello John, how are you?
    John->Alice: Great!
    Alice->John: See you later!
```

The others diagrams work correctly!

If it help's you, mark this comment as the answer 😄

**Comment by emilien-puget** [comment]
do you have an idea why this diagram, taken from the official doc https://mermaid.js.org/syntax/classDiagram.html#define-namespace doesnt work ?

```mermaid
classDiagram
namespace BaseShapes {
    class Triangle
    class Rectangle {
      double width
      double height
    }
}
```

**Comment by F4NT0** [comment, marked as answer]
Github don't support the namespace use on class diagrams, so it works without the namespace.
Github have some issues with the use of some parts of mermaid.js due to some configurations to complex structures like namespace and icons.

```mermaid
classDiagram
       class Triangle
       class Rectangle {
          double width
          double height
       }
```

**Reply by lat9** [reply]
At what point can we expect GitHub to support the mermaid namespace use with class diagrams?
